### ISSUE TITLE
"MCParticle" namespace collision with lcfiplus

### ISSUE THREAD

**Issue by vvolkl** [HEP-FCC/FCCAnalyses#138]
```
~/r/FCCAnalyses/examples (master)$ python FCCee/higgs/eeh/analysis.py /eos/experiment/fcc/ee/generation/DelphesEvents/spring2021/IDEA/p8_noBES_ee_H_Hcc_ecm125/events_016443522.root 
Warning in <TInterpreter::ReadRootmapFile>: class  edm4hep::ObjectID found in libedm4hepDict.so  is already in libedm4drDict.so 
Load cxx analyzers ... 
edm4hep   <cppyy.gbl.edm4hep.ReconstructedParticleData object at 0x8bb7170>
podio     <cppyy.gbl.podio.ObjectID object at 0x6c9b870>
fccana    <C++ overload "dummyLoader" at 0x7f83eef7b730>
 done
input_line_77:2:77: error: reference to 'MCParticle' is ambiguous
auto lambda0 = [](ROOT::VecOps::RVec<edm4hep::MCParticleData>& var0){return MCParticle::sel_genStatus(1)(var0)
                                                                            ^
/cvmfs/sw.hsf.org/spackages5/fccanalyses/0.3.7/x86_64-centos7-gcc11.2.0-opt/rqzll/include/FCCAnalyses/MCParticle.h:21:11: note: candidate found by name lookup is 'MCParticle'
namespace MCParticle{
          ^
/cvmfs/sw.hsf.org/spackages5/lcfiplus/0.10.1/x86_64-centos7-gcc11.2.0-opt/gzpnj/include/lcfiplus.h:651:7: note: candidate found by name lookup is 'lcfiplus::MCParticle'
class MCParticle : public TLorentzVector {
      ^
input_line_81:2:77: error: reference to 'MCParticle' is ambiguous
auto lambda0 = [](ROOT::VecOps::RVec<edm4hep::MCParticleData>& var0){return MCParticle::sel_genStatus(1)(var0)
                                                                            ^
/cvmfs/sw.hsf.org/spackages5/fccanalyses/0.3.7/x86_64-centos7-gcc11.2.0-opt/rqzll/include/FCCAnalyses/MCParticle.h:21:11: note: candidate found by name lookup is 'MCParticle'
namespace MCParticle{
          ^
/cvmfs/sw.hsf.org/spackages5/lcfiplus/0.10.1/x86_64-centos7-gcc11.2.0-opt/gzpnj/include/lcfiplus.h:651:7: note: candidate found by name lookup is 'lcfiplus::MCParticle'
class MCParticle : public TLorentzVector {
      ^
Traceback (most recent call last):
  File "/afs/cern.ch/user/v/vavolkl/repo/FCCAnalyses/examples/FCCee/higgs/eeh/analysis.py", line 112, in <module>
    analysis.run()
  File "/afs/cern.ch/user/v/vavolkl/repo/FCCAnalyses/examples/FCCee/higgs/eeh/analysis.py", line 33, in run
    df2 = (self.df
cppyy.gbl.std.runtime_error: Template method resolution failed:
  ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void> ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Define(basic_string_view<char,char_traits<char> > name, basic_string_view<char,char_traits<char> > expression) =>
    runtime_error: 
RDataFrame: An error occurred during just-in-time compilation. The lines above might indicate the cause of the crash
 All RDF objects that have not run an event loop yet should be considered in an invalid state.

  ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void> ROOT::RDF::RInterface<ROOT::Detail::RDF::RLoopManager,void>::Define(basic_string_view<char,char_traits<char> > name, basic_string_view<char,char_traits<char> > expression) =>
    runtime_error: 
RDataFrame: An error occurred during just-in-time compilation. The lines above might indicate the cause of the crash
 All RDF objects that have not run an event loop yet should be considered in an invalid state.
```

Workaround is to remove the lcfiplus headers from ROOT_INCLUDE_PATH.

The namespace here should probably be called `FCCAnalyses::MCParticle`.

**Comment by clementhelsens** [comment]
thanks for reporting @vvolkl , I will change the namespace consistently

**Comment by vvolkl** [comment]
Yes, I think putting all analyzers in a `FCCAnalyses::` namespace is a good idea anyway, but I also don't understand this error - The other candidate is in a namespace `lcfiplus::MCParticle` so I don't see where the ambiguity is?

**Comment by clementhelsens** [comment]
Indeed! not clear why the compiler complains about this ambiguous declaration.
I will start by moving now the ```MCParticle``` to  ```FCCAnalyses::MCParticle``` such that we don't have to remove headers from ```ROOT_INCLUDE_PATH```

**Comment by clementhelsens** [comment]
this is fixed (only for MCParticles) in #142
